src/compiler/crystal/types.cr

```diff
@@ -593,7 +593,7 @@ module Crystal
       0
     end
 
-    def type_desc
+    def type_desc(indefinitive_article = false)
```

Comment:
I don't think this method is ever invoked in the error messages with `type_desc(true)` but I'm not sure. Does anyone know for sure?
If it is, then an algorithm has to be made which determines whether "a" or "an" needs to be in front of the word and that could be quite complicated because this type could theoretically be anything, any word. But then the errors should probably just be reordered to avoid the "a".